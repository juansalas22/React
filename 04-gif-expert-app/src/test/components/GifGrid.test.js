import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'Prueba';

    test('Debe mostar <GifGrid /> correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    })

    test('Debe mostar <GifGrid /> correctamente', () => {

        const gifs = [{
            id:'ABX',
            url: 'Web',
            title: 'probando'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })

})
