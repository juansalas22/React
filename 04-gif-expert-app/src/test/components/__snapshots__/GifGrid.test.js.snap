// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Prueba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostar <GifGrid /> correctamente 2`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Prueba
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABX"
      key="ABX"
      title="probando"
      url="Web"
    />
  </div>
</Fragment>
`;
